src/Internal/UriValidation/UriValidation.cpp

```diff
@@ -0,0 +1,14 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include "UriValidation.h"
+
+namespace AppInstaller::UriValidation
+{
+    UriValidationResult UriValidation(const std::string&)
+    {
+        // In Dev mode, allow all URIs

```

Comment:
I'm a bit confused by this comment, can you help me understand it? The way it reads to me at the moment is that because you're currently testing you want to allow all URI's?

This new functionality may also benefit from having test hooks (similar to the Zip file malware scan) if it isn’t already needed for the tests John mentioned

https://github.com/microsoft/winget-cli/blob/2306d08b6f2ed2b6d23a3442258e1f2067c13b37/src/AppInstallerCommonCore/Archive.cpp#L52-L68

Reply:
The actual SmartScreen logic/code will be injected in the release build pipeline and this method will be substituted. I plan on adding tests later. John mentioned to me offline that there are test hooks I can leverage, but I haven't had the chance to explore them yet. Thank you for providing an example, I will consider that once I add some tests.